JamGame/Assets/Scripts/DataProviderServiceLocator.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using Common;
+using UnityEngine;
+
+public class DataProviderServiceLocator : MonoBehaviour
```

Comment:
Probably it woud better implement traditional Unity singleton for more control, because unity don't good works with c# constructors and destructors. Register and unregister execute in enable and disable.

Reply:
DataProvider is *not* a `MonoBehaviour`